From 4d922895addf81586308bffaea9ff7e0106eba48 Mon Sep 17 00:00:00 2001
From: Andriy Grytsenko <andrej@rep.kiev.ua>
Date: Mon, 15 Jun 2015 14:54:48 +0300
Subject: [PATCH] [SF#758] Fix crash in taskbar plugin after number of desktops
 was updated.

The pointer to invalid menu item was not reset, what's a shame.
---
 ChangeLog               | 2 ++
 plugins/launchtaskbar.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 1390eac..bef51b8 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,5 @@
+* Fixed crash in taskbar plugin after number of desktops was updated.
+
 0.8.1
 -------------------------------------------------------------------------
 * Improved OSS headers detection a bit.
diff --git a/plugins/launchtaskbar.c b/plugins/launchtaskbar.c
index 062e149..a1b4fb9 100644
--- a/plugins/launchtaskbar.c
+++ b/plugins/launchtaskbar.c
@@ -3348,6 +3348,8 @@ static void taskbar_make_menu(LaunchTaskBarPlugin * tb)
     /* Deallocate old menu if present. */
     if (tb->menu != NULL)
         gtk_widget_destroy(tb->menu);
+    /* The pointer to menu became invalid, reset it now. */
+    tb->workspace_menu0 = NULL;
 
     /* Allocate menu. */
     GtkWidget * menu = gtk_menu_new();
-- 
2.1.3

